packages/core/src/external/aws/lambda-logic/bundle-to-brief.ts

```diff
@@ -731,7 +949,135 @@ function replaceReferencesWithData(
+
+  const sortedLabReports = labReports.sort((a, b) => {
+    const aDates = getDatesFromEffectiveDateTimeOrPeriod(a);
+    const bDates = getDatesFromEffectiveDateTimeOrPeriod(b);
+
+    const aDate = aDates.find(d => d !== undefined);
+    const bDate = bDates.find(d => d !== undefined);
+
+    if (!aDate) return 1;
+    if (!bDate) return -1;
+    return bDate.localeCompare(aDate);
+  });
+
+  const recentLabReports = sortedLabReports.slice(0, NUM_MOST_RECENT_LABS_TO_KEEP);
+
+  return [...recentLabReports, ...nonLabReports];
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function filterReportsByPerformerAndCategory(reports: any[] | undefined): any[] | undefined {
+  if (!reports) return undefined;
+  const MAX_REPORTS_PER_GROUP = 3;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const reportGroups = new Map<string, any[]>();
+
+  reports.forEach(report => {
+    const performer = report.performer ?? "unknown";
+    const type = report.type ?? "unknown";
+    const key = `${performer}|${type}`;
+
+    if (!reportGroups.has(key)) {
+      reportGroups.set(key, []);
+    }
+    reportGroups.get(key)?.push(report);
+  });
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const filteredReports: any[] = [];
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const garbageCollector: any[] = [];
+
+  reportGroups.forEach(group => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const sortedGroup = group.sort((a: any, b: any) => {
+      const aDates = getDatesFromEffectiveDateTimeOrPeriod(a);
+      const bDates = getDatesFromEffectiveDateTimeOrPeriod(b);
+
+      const aDate = aDates.find(d => d !== undefined);
+      const bDate = bDates.find(d => d !== undefined);
+
+      if (!aDate) return 1;
+      if (!bDate) return -1;
+      return bDate.localeCompare(aDate);
+    });
+
+    filteredReports.push(...sortedGroup.slice(0, MAX_REPORTS_PER_GROUP));
+    garbageCollector.push(...sortedGroup.slice(MAX_REPORTS_PER_GROUP));
+  });
+
+  return filteredReports;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function filterOutDuplicateReports(reports: any[] | undefined): any[] | undefined {
+  const formDataSet = new Set<string>();
+  return reports?.filter(entry => {
+    if (entry.presentedForm) {
+      const newPresentedForm: string[] = [];
+      entry.presentedForm.forEach((text: string) => {
+        const sentences = text
+          .split(/[.\n]/)
+          .map(s => s.trim())
+          .filter(s => s.length > 0);
+
+        const filteredSentences = sentences.filter(sentence => {
+          if (formDataSet.has(sentence)) {
+            return false;
+          }
+          formDataSet.add(sentence);
+          return true;
+        });
+        newPresentedForm.push(filteredSentences.join(". "));
+      });
+
+      const filsss = newPresentedForm.join("\n");
```

Comment:
prob a typo on `filsss`?

Reply:
oh crap, yes.. sometimes i use random strings for var names when im not sure what im gonna name them.. but this time i forgot to update it 😅